src/expr/subs.cpp

```diff
@@ -69,13 +70,21 @@ void Subs::add(const std::vector<Node>& vs)
   }
 }
 
-void Subs::add(Node v, Node s)
+void Subs::add(const Node& v, const Node& s)
 {
-  Assert(s.isNull() || v.getType().isComparableTo(s.getType()));
+  if (Configuration::isAssertionBuild())
```

Comment:
Note that this introduces a performance overhead because `isAssertionBuild()` will not be inlined / optimized away.
How about we keep the assertion, and instead override `add()` in the arithmetic substitution?

Reply:
yeah i had considered this, I'll change.